### PR TITLE
library: Add vala demo for Scale

### PR DESCRIPTION
**Pull request by SpikedPaladin**
Adds Vala code demo for Scale

### CODE REVIEW

File: src/Library/demos/Scale/main.vala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    var scale_value = scale_one.get_value ();
+    if (scale_value == scale_one.adjustment.upper) {
+      stdout.printf ("Maximum value reached");
+    } else if (scale_value == scale_one.adjustment.lower) {
+      stdout.printf ("Minimum value reached");
+    }
+  });
+
+  scale_two.value_changed.connect(() => {
+    var scale_value = scale_two.get_value ();
```
Comment (lw64) [inline src/Library/demos/Scale/main.vala:28]: I wouldn't use var here. What type is this?

File: src/Library/demos/Scale/main.vala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    scale_value /= 50;
+    if (scale_value.to_string().contains(".")) return;
```
Comment (lw64) [inline src/Library/demos/Scale/main.vala:30]: This line is not very clear. What is the code trying to do here? Checking if scale_value is a whole number? Either put a comment or use another method that makes it better understandable.

**Review by lw64 — CHANGES_REQUESTED** [review]
Everything else looks alright!

**Comment by SpikedPaladin** [comment]
@lw64 updated

**Pull request merged**